app/src/main/java/com/github/muellerma/prepaidbalance/ui/MainActivity.kt

```diff
@@ -131,17 +142,21 @@ class MainActivity : AbstractBaseActivity(), SwipeRefreshLayout.OnRefreshListene
     private fun exportAsCsv() {
         launch {
             val content = buildCsv()
-
+            val filename = "prepaid-balance-${System.currentTimeMillis()}.csv"
             try {
-                val filename = "prepaid-balance-${System.currentTimeMillis()}.csv"
                 writeToFileInDownloads(content, filename)
                 showSnackbar(getString(R.string.export_saved_file, filename))
                 return@launch
-            } catch (e: Exception) {
+            } catch (e: FileNotFoundException) {
+                if (hasPermissions(READ_EXTERNAL_STORAGE) && hasPermissions(WRITE_EXTERNAL_STORAGE)) {
+                    writeToFileInDownloads(content, filename)
+                    showSnackbar(getString(R.string.export_saved_file, filename))
```

Comment:
Why do you call `writeToFileInDownloads()` again? It failed above and nothing has changed, so it will fail again.

Reply:
Some API version needs permission before you can read and write to storage. That is why there is this catch line. `catch (e: FileNotFoundException)`.
Alternatively I could check for build and handle the API exception 

This is the apk you asked for. It is in the zip file 
[prepaiddebug.zip](https://github.com/mueller-ma/PrepaidBalance/files/10951554/prepaiddebug.zip)

Reply:
@brotzeller Can you test this zip?

Reply:
If the app already has read and write permissions and `writeToFileInDownloads()` in line 147 throws `FileNotFoundException`, `writeToFileInDownloads()` is called again in line 152 without changing anything before. What's the purpose of this second call?

> Some API version needs permission before you can read and write to storage. 

Did you find some documentation on this? All I found says that the downloads folder is writable without permissions.